x-pack/plugins/security_solution/server/endpoint/routes/policy/handlers.ts

```diff
@@ -31,3 +35,34 @@ export const getHostPolicyResponseHandler = function (
     }
   };
 };
+
+export const getAgentPolicySummaryHandler = function (
+  endpointAppContext: EndpointAppContext
+): RequestHandler<undefined, TypeOf<typeof GetAgentPolicySummaryRequestSchema.query>, undefined> {
+  return async (context, request, response) => {
+    try {
+      const result = await getAgentPolicySummary(
+        endpointAppContext,
+        context.core.savedObjects.client,
+        request.query.package_name,
+        request.query?.policy_id || undefined
+      );
+      const responseBody = {
+        package: request.query.package_name,
+        versions_count: { ...result },
+      };
+
+      const body: GetAgentSummaryResponse = {
+        summary_response: request.query?.policy_id
+          ? { ...responseBody, ...{ policy_id: request.query?.policy_id } }
+          : responseBody,
+      };
+
+      return response.ok({
+        body,
+      });
+    } catch (err) {
+      return response.internalError({ body: err });
```

Comment:
I don't belive the stack trace makes it over the API response. May want to consider logging the `err` here for debuging purposes